Polyhedron/demo/Polyhedron/include/CGAL/IO/read_surf_trianglemesh.h

```diff
@@ -158,20 +159,23 @@ void read_surf(std::istream& input, std::vector<Mesh>& output, std::vector<Mater
       iss.clear();
       iss.str(line);
       iss >> index[0] >> index[1] >> index[2];
+      CGAL::cpp11::array<vertex_descriptor, 3> face;
       for(int id=0; id<3; ++id)
       {
-        if(vertices.find(index[id]) == vertices.end())
+        if(vertices[index[id]] == null_vertex)
         {
-          vertices.insert(std::make_pair(index[id], add_vertex(points[index[id]-1], mesh)));
+          vertices[index[id]] = add_vertex(points[index[id]-1], mesh);
         }
+        face[id] = vertices[index[id]];
       }
-      std::vector<vertex_descriptor> face;
-      for(int id=0; id<3; ++id)
-        face.push_back(vertices[index[id]]);
       CGAL::Euler::add_face(face, mesh);
     }
-    output.push_back(mesh);
-  }
+    // reset the `vertices` vector, unless that was the last iteration of
+    // the loop
+    if(++i < nb_patches) std::fill(vertices.begin(),
+                                   vertices.end(),
+                                   null_vertex);
```

Comment:
And at the end of each iterations, the vector `vertices` is filled again with the null vertex descriptor.